### PR TITLE
Packaging: simplify building an installable package for Windows

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

This is the continuation of [PR 5499 ](https://github.com/PowerShell/PowerShell/pull/5499) that had to be abandoned due to a fatal merge conflict and I did not want to risk accidentally reverting recent fixes.

- Remove unnecessary/unused default for productGuid because it always gets a new Guid when being called from `Start-PSPackage`
- Add defaults for required files but also add extra path validation attribute
- Rename ProductGuid to ProductCode

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -2052,6 +2052,16 @@ function Get-PackageVersionAsMajorMinorBuildRevision
     $packageVersion
 }
 
+<#
+    .Synopsis
+        Creates a Windows installer MSI package and assumes that the binaries are already built using 'Start-PSBuild'.
+        This only works on a Windows machine due to the usage of WiX.
+    .EXAMPLE
+        # This example shows how to produce a Debug-x64 installer for development purposes.
+        cd $RootPathOfPowerShellCheckout
```
Comment (SteveL-MSFT) [inline build.psm1]: Is it better to use the term `Repo` instead of `Checkout`?

File: build.psm1 [hunk 2]
```diff
@@ -2070,8 +2080,9 @@ function New-MSIPackage
         [string] $ProductVersion,
 
         # Product Guid needs to change for every version to support SxS install
```
Comment (SteveL-MSFT) [inline build.psm1]: This comment should probably be changed to: Product Code needs to be different for every version

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -243,7 +243,7 @@ function Start-PSPackage {
                     AssetsPath = "$PSScriptRoot\..\..\assets"
                     LicenseFilePath = "$PSScriptRoot\..\..\assets\license.rtf"
                     # Product Guid needs to be unique for every PowerShell version to allow SxS install
```
Comment (SteveL-MSFT) [inline tools/packaging/packaging.psm1]: This comment should probably be changed to: Product Code needs to be different for every version

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by bergmeister** [comment]
@SteveL-MSFT OK, I addressed your comments.

File: build.psm1 [hunk 4]
```diff
@@ -2069,9 +2079,10 @@ function New-MSIPackage
         [ValidateNotNullOrEmpty()]
         [string] $ProductVersion,
 
-        # Product Guid needs to change for every version to support SxS install
+        # Product Code needs to change for every version to support SxS install
```
Comment (TravisEz13) [inline build.psm1]: Does this comment make sense any more?

`The ProductCode property is a unique identifier for the particular product release`

from https://msdn.microsoft.com/en-us/library/windows/desktop/aa370854(v=vs.85).aspx

File: tools/packaging/packaging.psm1 [hunk 5]
```diff
@@ -242,8 +242,8 @@ function Start-PSPackage {
                     ProductVersion = $Version
                     AssetsPath = "$PSScriptRoot\..\..\assets"
                     LicenseFilePath = "$PSScriptRoot\..\..\assets\license.rtf"
-                    # Product Guid needs to be unique for every PowerShell version to allow SxS install
-                    ProductGuid = New-Guid
+                    # Product Code needs to be unique for every PowerShell version to allow SxS install
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: This is not correct either...  but here just remove the `to allow SxS install` part.

File: assets/Product.wxs [hunk 6]
```diff
@@ -5,7 +5,7 @@
   <!-- TBD:Point to the actual release -->
   <?define InfoURL="https://github.com/PowerShell/PowerShell" ?>
   <?define ProductName = "$(env.ProductName)" ?>
-  <?define ProductGuid = "$(env.ProductGuid)" ?>
+  <?define ProductCode = "$(env.ProductCode)" ?>
   <!-- UpgradeCode GUID MUST REMAIN SAME THROUGHOUT ALL VERSIONS, otherwise, updates won't occur. -->
```
Comment (TravisEz13) [inline assets/Product.wxs]: This is actually where the comment about SxS should be.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
I have some comments about the comments

**Review by TravisEz13 — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**